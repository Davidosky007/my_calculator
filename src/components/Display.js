/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledDisplay = styled.h2`
display: flex;
color: white;
font-weight: bold;
float: right;
`;

const StyledDisCompo = styled.section`
background-color: gray;
height: 100px;
padding: 15px;
width: 400px;
margin: 0 auto;
border-radius: 30px;
`;

const Display = (props) => {
  const { result } = props;
  return (
    <StyledDisCompo>
      <StyledDisplay>
        {result}
      </StyledDisplay>
    </StyledDisCompo>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
