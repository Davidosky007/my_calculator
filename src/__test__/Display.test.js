/* eslint-disable linebreak-style */
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import Display from '../components/Display';

describe('test Display component', () => {
  test('render Display component properly', () => {
    const display = create(<Display />).toJSON();
    expect(display).toMatchSnapshot();
  });
  test('it renders Display component with the text 0', () => {
    render(<Display />);
    expect(screen.getByText(0)).toBeInTheDocument();
  });
});
