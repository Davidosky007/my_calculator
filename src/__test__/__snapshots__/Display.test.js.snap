// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Display component render Display component properly 1`] = `
<section
  className="sc-gtsrHT ilhZDh"
>
  <h2
    className="sc-bdnxRM gNjEKx"
  >
    0
  </h2>
</section>
`;
